shell/platform/embedder/embedder.h

```diff
@@ -1740,17 +1760,30 @@ typedef struct {
   /// immediately after the root isolate has been created and marked runnable.
   VoidCallback root_isolate_create_callback;
   /// The callback invoked by the engine in order to give the embedder the
+  /// chance to respond to updates to semantics nodes and custom actions from
+  /// the Dart application.
+  ///
+  /// The callback will be invoked on the thread on which the `FlutterEngineRun`
+  /// call is made.
+  ///
+  /// If this callback is provided, update_semantics_node_callback and
+  /// update_semantics_custom_action_callback must not be provided.
+  FlutterUpdateSemanticsCallback update_semantics_callback;
+  /// The old callback invoked by the engine in order to give the embedder the
   /// chance to respond to semantics node updates from the Dart application.
   /// Semantics node updates are sent in batches terminated by a 'batch end'
   /// callback that is passed a sentinel `FlutterSemanticsNode` whose `id` field
   /// has the value `kFlutterSemanticsNodeIdBatchEnd`.
   ///
   /// The callback will be invoked on the thread on which the `FlutterEngineRun`
   /// call is made.
+  ///
+  /// Embedders should use `update_semantics_callback` instead. If this callback
+  /// is provided, `update_semantics_callback` must not be provided.
   FlutterUpdateSemanticsNodeCallback update_semantics_node_callback;
-  /// The callback invoked by the engine in order to give the embedder the
+  /// The old callback invoked by the engine in order to give the embedder the
```

Comment:
`old` in relative to what? may be use the the word `legacy`

Reply:
Sounds good, replaced `old` with `legacy` everywhere.